server/jms/src/main/java/org/hawkular/apm/server/jms/CommunicationDetailsDeriverMDB.java

```diff
@@ -49,7 +48,7 @@
     private TracePublisherJMS tracePublisher;
 
     @Inject
-    private CommunicationDetailsPublisher communicationDetailsPublisher;
+    private CommunicationDetailsPublisherJMS communicationDetailsPublisher;
```

Comment:
The reason is mentioned in the reviewer notes - it is like there will be atleast one other concrete implementation of the CommunicationDetailsPublisher soon (Kafka) so wanted to ensure that within the MDB/JMS context, only the JMS publisher implementation is chosen.